event_sourcery/event_store/subscription.py

```diff
@@ -41,38 +44,43 @@ def to_events(self, events: list[Type[Event]]) -> Builder:
         ...
 
 
-class Positioner(abc.ABC):
+class Subscriber(abc.ABC):
+    in_transaction: ContextManager[Iterator[Entry]]
+
     @abc.abstractmethod
-    def __call__(self, start_from: Position) -> Kind:
+    def start_from(self, position: Position) -> Filter:
         ...
 
 
 @dataclass(repr=False)
-class Engine(Kind, Builder):
-    start_from: Position
-    strategy: SubscriptionStrategy
-    serde: Serde
+class Engine(Subscriber, Filter, Builder):
```

Comment:
I still have trouble to wrap my head around those `Engine` classes. One is used here and the other one is result of a method of `EventStoreFactory`.

I don't have a better name for THIS `Engine` yet, but the second one could be perhaps called `Backend`?

Reply:
Got this one. 

We're addressing `Interface Segregation Principle` here (in a way). I wasn't able to find any applicable piece of advice regarding Builder + Fluent Interface that was meant to separate interfaces in a way we do here.

What we (we? you?) did here was to "invent" a concept of phases to guide a user of the library to create a `Subscription` in a specific way. I think we miss `Phase` suffix to each ABC name to make this concept explicit.

`Builder` -> `BuildPhase`
`Filter` (interestingly it's not Filterer) -> `FilterPhase`
`Subscription` (formerly `Subscriber`) -> `PositionPhase` 

Then, `Engine` can be named `SubscriptionBuilder`.

I used ChatGPT to write me a solution to this. It was completely wrong, but while I was writing 3rd prompt I realised we missed "Phase" concept.

Reply:
done